pkg/gcp/actuator/utils_test.go

```diff
@@ -0,0 +1,39 @@
+package actuator
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestCalculateSliceDiff(t *testing.T) {
+	tests := []struct {
+		name            string
+		original        []string
+		new             []string
+		expectedAdded   []string
+		expectedRemoved []string
+	}{
+		{
+			name:            "basic",
+			original:        []string{"a", "b", "c"},
+			new:             []string{"a", "c", "d"},
+			expectedAdded:   []string{"d"},
+			expectedRemoved: []string{"b"},
+		},
+		{
+			name:            "extreme",
+			original:        []string{"a", "b", "c", "d", "e", "f"},
+			new:             []string{"a", "c", "d", "g", "h", "q", "z"},
+			expectedAdded:   []string{"q", "z", "h", "g"},
+			expectedRemoved: []string{"f", "e", "b"},
+		},
```

Comment:
I think it would be more concise to test the following:

- There are no differences
- There are expectedAdded
- There are expectedRemoved
- There are expectedAdded and expectedRemoved

and also situations where
- original is empty
- new is empty